File: CodingPracticeCPP/luogu/P1059 明明的随机数/P1059 明明的随机数 桶排序版.cpp
// Author: FrancisDShanks
// Email: tctcdtc@gmail.com
#include <iostream>
using namespace std;

int main(){
	int n,tmp,cnt=0;
	cin>>n;
	bool bucket[1000]={0}; // use bool type is enough 
	for(int i=0;i<n;i++){
		cin>>tmp;
		if(!bucket[tmp]) cnt+=1;
		bucket[tmp] = true;
	}
	
	cout<<cnt<<endl;
	for(int i=1;i<1001;i++){
		if(bucket[i]) cout<<i<<" ";
	}
	
	return 0;
}

File: CodingPracticeCPP/luogu/P1059 明明的随机数/P1059 明明的随机数 桶排序版.py
# Author: FrancisDShanks
# Email: tctcdtc@gmail.com

if __name__ == "__main__":
    n = int(input())
    a = [False] * 1001
    cnt = 0
    tmp = [int(i) for i in input().strip().split()]
    for i in range(len(tmp)):
        if not a[tmp[i]]:
            cnt += 1
        a[tmp[i]] = True
    print(cnt)
    for i in range(1,1001):
        if a[i]:
            print(i,end=' ')
        
